### PR TITLE
Add CDN Hacker News API, move site onto Firebase Hosting

### PR DESCRIPTION
**Pull request by davideast**
**tl;dr**
This PR adds in an edge cached Hacker News API to hnpwa.com using Firebase Hosting + Firebase Functions. Hosting infrastructure moved to Firebase Hosting from AppEngine.

**Motivation**
Drop latency of API calls. The idea here is to provide a single API for each HNPWA to use. The Firebase HNPWA API is edge cached, which can result if sub 100ms response times around the world. If each app uses the API the edge caches will always be warm and keep response times low for faster page loads.

**CI**
When this PR is merged, we'll need to make sure that the Travis deployment actually goes through.

### CODE REVIEW

File: site/.firebaserc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "projects": {
+    "default": "hnpwa-site"
```
Comment (addyosmani) [inline site/.firebaserc]: Does Firebase hosting support adding permissions for multiple folks to access config for this FB project through the console? Asking as it would be good to know if we'll need to chase you for help each time :)

Comment (davideast) [inline site/.firebaserc]: Yep! Same IAM permissions as GCP. I can easily add anyone or we can create an admin account as the owner.

**Comment by davideast** [comment]
@housseindjirdeh I have access to the registrar. I've mirrored the DNS records of the current CloudFlare site to Firebase Hosting. It should take about 12 hours or so until we see it take effect.
